03_module/trouble_module.c

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+/*
+ * Trouble kernel module, which returns
+ * fail or success based on random rand_number
+ */
+
+
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/printk.h>
+#include <linux/moduleparam.h>
+#include <linux/random.h>
+
+
+MODULE_AUTHOR("Denys Andreichuk <Denys.Andreichuk@globallogic.com");
+MODULE_DESCRIPTION("Module");
+MODULE_LICENSE("Dual BSD/GPL");
+
+#define SUCCESS_RATE_PERCENT 50 /* percentage of successfull module loads */
+
+static uint rand_number;
+module_param(rand_number, uint, 0);
+
+static int __init da_mod_init(void)
+{
+	uint ret_value = 1;
+
+	pr_info("Trouble module trying to start\n");
+
+	get_random_bytes(&rand_number, 1);
```

Comment:
Please remove this line.
task 4 is related to module parameter. and output  dependent on this parameter.
In this case you are able to transfer this parameter into module as "insmod trouble_module.ko rand_number=60" or "insmod trouble_module.ko rand_number=20"